pennylane/operation.py

```diff
@@ -1782,6 +1784,8 @@ def __mul__(self, a):
         r"""The scalar multiplication operation between a scalar and an Observable/Tensor."""
         if isinstance(a, (int, float)):
             return qml.Hamiltonian([a], [self], simplify=True)
+        if inspect.isfunction(a):
```

Comment:
I thought we settled on `isfunction` to avoid it using this method to try to resolve `ParametrizedHamiltonian * Observable`. This `__mul__` method is written to multiply a scalar, or now, a callable that can return a scalar if given parameters, onto an Observable. Its behaviour is nonsensical if it accepts a `ParametrizedHamiltonian` as `a`; using `isfunction` avoids this, and `iscallable` does not.

Reply:
Okay, I missed the last part of the discussion on the thread. Tomorrow I'll put it back to `callable` and just add a clause dealing specifically with `ParametrizedHamiltonian` :)